### PR TITLE
Lennonjohto varaustilanne

### PR DESCRIPTION
**Pull request by hamidaebadi**
Related to Issue #17 Lennonjohtaja voi nähdä varaustilanteen

### What changes has been added?
I've done a few changes in both backend and frontend. 
**Backend**
- Handling get-request to api/flight-staff/reservation-status. This handler gets data from database and return as a json to the client.
- Response contains data about current reservation status including both available and reserved time slots.

**Frontend**
- A new page (staff.tsx) added onto the page directory.
- staff.tsx uses React Query to fetch reservation status data from server and manage them in the Staff component.

### Expected Behaviour

- We can navigate to the lennonjohto-page using main page's navigation bar
- Lennonjohto-page shows the current data about reservation including both available and reserved time slots in a simple unordered list
- As we have no test data in our database, the page should only show the following content
Screenshot 2023-02-01 at 12 50 02

### CODE REVIEW

**Comment by antonlep** [comment]
image
 For available timeslot, there could be additional info which shows how many time slots are free. In that case max available times are 2, reservations are 1, so available times are 1. 

Reservations are strored in Reservation table instead of ReservedTimeslot, that could be used instead of ReservedTimeslot. For that table could be included connection to User table, so related user is shown.

**Comment by hamidaebadi** [comment]
@antonlep 
I've added a field "freeSlotsAmount" showing the number of available slots from a specific time.

File: backend/src/index.ts [hunk 1]
```diff
@@ -21,5 +21,39 @@ app.delete('/api/timeslots/:id', async (req: express.Request, res: express.Respo
...
+  try {
+    const returnedTimeSlots = await Timeslot.findAll({
+      include: {
+        model: Reservation,
+        attributes: ['startTime', 'endTime', 'info'],
+      },
+    });
+    const availableSlots = returnedTimeSlots.map((obj) => (
+      {
+        ...obj,
```
Comment (antonlep) [inline backend/src/index.ts:38]: this should probably be ...obj.dataValues,
so resulting object is in correct format

Comment (hamidaebadi) [reply backend/src/index.ts]: Oops!
Sorry for typo
It should be correct now

**Review by antonlep — APPROVED** [review]
Looks good!

**Comment by hamidaebadi** [comment]
@antonlep could you please review my codes again?
Some refactoring done

File: backend/src/index.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import express from 'express';
 import cors from 'cors';
 import timeslotRouter from './routes/timeslots';
+import flightStaggRouter from './routes/flightstaff';
```
Comment (vaisest) [inline backend/src/index.ts:4]: Kirjoitusvirhe?

Comment (vaisest) [reply backend/src/index.ts:4]: Muutenkin flight staff on vähän epäselvä? Eikö kyseessä ole enemmänkin lennonjohto eli joku "flight control"?

File: backend/src/routes/flightstaff.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import express from 'express';
+import flightstaffService from '../services/flightstaffService';
+
+const router = express.Router();
+
+router.get('/reservation-status', async (_req: any, res: express.Response) => {
+  try {
+    const reservationStatus = await flightstaffService.getReservationStatus();
+    res.send(JSON.stringify(reservationStatus));
```
Comment (vaisest) [inline backend/src/routes/flightstaff.ts:9]: `res.json()`
https://expressjs.com/en/api.html#res.json

File: frontend/src/pages/Management.tsx [hunk 4]
```diff
@@ -1,9 +1,47 @@
...
 function Management() {
+  const { data, isLoading, isError } = useQuery(QueryKeys.ReservationStatus, getResrvationStatus, {
+    refetchOnWindowFocus: false,
+  });
   return (
-    <div>
-      <p className="text-3xl">Management Page</p>
+    <div className="flex flex-col space-y-2">
+      <h1 className="text-3xl">Management</h1>
+      {isError && <p className="text-base">Erro while trying to fetch reservation status</p>}
```
Comment (vaisest) [inline frontend/src/pages/Management.tsx:14]: Kirjoitusvirhe: error

File: frontend/src/pages/Management.tsx [hunk 5]
```diff
@@ -1,9 +1,47 @@
...
+                    {data.availableSlots.map(
+                      (slot: any) => <li key={slot.id}>{slot.startTime}</li>,
+                    )}
+                  </ul>
+                )}
+            </div>
+            <div>
+              <h1>Varatut Ajat</h1>
+              {data.reservedSlots.length === 0
+                ? <p> Ei Varattuja aikoja ole olemassa</p>
```
Comment (vaisest) [inline frontend/src/pages/Management.tsx:34]: Kirjoitusvirhe. Varmaan "Ei varattuja aikoja"?

Muutenkin nyt englantia ja suomea sekaisin mikä on vähän outoa

**Review by vaisest — CHANGES_REQUESTED** [review]
En kokeillut toiminnallisuuta ollenkaan. Katsoin vain muutokset läpi

File: backend/src/index.ts (hunk 2 above)
Comment (hamidaebadi) [reply backend/src/index.ts:4]: Joo kirjoitusvirhe!
käytetään flightControl nimeä

File: backend/src/routes/flightstaff.ts (hunk 3 above)
Comment (hamidaebadi) [reply backend/src/routes/flightstaff.ts:9]: ok

File: frontend/src/pages/Management.tsx (hunk 5 above)
Comment (hamidaebadi) [reply frontend/src/pages/Management.tsx:34]: Korjattu!

**Review by vaisest — APPROVED** [review]

**Pull request merged**